File: 백준/Silver/10845. 큐/큐.kt
package org.example

import java.util.LinkedList

fun main() {
    val br = System.`in`.bufferedReader()
    val n = br.readLine().toInt()
    val queue = LinkedList<String>()
    val sb = StringBuilder()
    repeat(n) {
        val command = br.readLine().split(" ")
        when(command[0]) {
            "push" -> {
                queue.add(command[1])
            }
            "pop"->{
                if(queue.isNotEmpty()){
                    sb.append("${queue.poll()}\n")
                }else{
                    sb.append("-1\n")
                }
            }
            "size"->{
                sb.append("${queue.size}\n")
            }
            "empty"->{
                if(queue.isEmpty()){
                    sb.append("1\n")
                }else{
                    sb.append("0\n")
                }
            }
            "front"->{
                if(queue.isEmpty()){
                    sb.append("-1\n")
                }else{
                    sb.append("${queue.first()}\n")
                }
            }
            "back"->{
                if(queue.isEmpty()){
                    sb.append("-1\n")
                }else{
                    sb.append("${queue.last()}\n")
                }
            }
        }

    }
    print(sb)

}

File: 백준/Silver/10845. 큐/큐.py
import sys
from collections import deque
input = sys.stdin.readline

n = int(input())
deque = deque()
ret = []
for _ in range(n):
    command = input().strip()
    if command.startswith("push"):
        _, num = command.split()
        deque.append(num)
    elif command.startswith("pop"):
        if deque:
            ret.append(deque.popleft())
        else:
            ret.append("-1")
    elif command.startswith("size"):
        ret.append(f"{len(deque)}")
    elif command.startswith("empty"):
        if deque:
            ret.append("0")
        else:
            ret.append("1")
    elif command.startswith("front"):
        if deque:
            ret.append(deque[0])
        else:
            ret.append("-1")
    elif command.startswith("back"):
        if deque:
            ret.append(deque[-1])
        else:
            ret.append("-1")

print('\n'.join(ret))